### ISSUE TITLE
When using Auth0 for OIDC, there's a CORS error

### ISSUE THREAD

**Issue by mraible** [jhipster/generator-jhipster-react-native#535]
**Describe the bug**

When you configure Auth0 as the identity provider, there's a CORS error when you try to login.

<img width="1296" alt="Screen Shot 2021-11-10 at 10 35 41 AM" src="https://user-images.githubusercontent.com/17892/141143385-999a2e7b-3a46-416e-8728-68c0372555f1.png">

**Instructions to reproduce**

https://github.com/mraible/mobile-jhipster/blob/main/demo.adoc#use-auth0-for-identity

**How to fix**

One thing I've found that works is to trim off the last / of the issuer. In `login.sagas.js`:

```js
export function* login(api) {
  try {
    const authInfo = yield select(selectAuthInfo);
    if (authInfo === null) {
      yield put(AuthInfoActions.authInfoRequest());
    }
    const { issuer, clientId } = authInfo;
    let issuerWithoutSlash = issuer;
    if (issuer.endsWith('/')) {
      issuerWithoutSlash = issuer.substring(0, issuer.length - 1);
    }
    const { accessToken, idToken } = yield call(doOauthPkceFlow, AppConfig.nativeClientId || clientId, issuerWithoutSlash);
    if (accessToken) {
      yield call(api.setAuthToken, accessToken);
      yield put(LoginActions.loginSuccess(accessToken, idToken));
      yield put(AccountActions.accountRequest());
      yield put({ type: 'RELOGIN_OK' });
    }
  } catch (e) {
    let errorMessage = e.type === 'Dismiss' ? 'Login Modal dismissed' : 'Login failed, please try again';
    yield put(LoginActions.loginFailure(errorMessage));
  }
}

// attempts to logout
export function* logout(api) {
  yield call(api.removeAuthToken);
  yield put(AccountActions.accountReset());
  yield put(AccountActions.accountRequest());
  const { clientId, issuer } = yield select(selectAuthInfo);
  const idToken = yield select(selectIdToken);
  let issuerWithoutSlash = issuer;
  if (issuer.endsWith('/')) {
    issuerWithoutSlash = issuer.substring(0, issuer.length - 1);
  }
  if (idToken) {
    yield call(logoutFromIdp, AppConfig.nativeClientId || clientId, issuerWithoutSlash, idToken);
  }
  yield put(LoginActions.logoutSuccess());
  yield put({ type: 'RELOGIN_ABORT' });
}
```

This will make it so login works, and it solves the CORS error on logout. However, it doesn't solve the 404 on logout.

<img width="1296" alt="Screen Shot 2021-11-10 at 10 38 50 AM" src="https://user-images.githubusercontent.com/17892/141143981-0631287d-5c56-49fe-a7d9-7f078cf834fd.png">

If I run my app in iOS Simulator, it seems this change is necessary as well. On iOS, logout _seems_ to work, but it doesn't. If I log out, then try to log in again, I'm not prompted to log in again. This seems to indicate logout is not working. 

I believe this could be because Auth0's well-known endpoint for OIDC doesn't have an `end_session_endpoint`:

https://jhipster.us.auth0.com/.well-known/openid-configuration

If you compare this to an Okta tenant, you'll see there is an `end_session_endpoint`:

https://dev-2530788.okta.com/oauth2/default/.well-known/openid-configuration

Keycloak has one too:

http://localhost:9080/auth/realms/jhipster/.well-known/openid-configuration

**Comment by ruddell** [comment]
I added the trimming of the last `/` if present on the URL.  I also prevented opening the logout modal if the end_session_endpoint does not exist.

Full logout only works smoothly on web.  It seems Expo's auth proxy doesn't support it.  There's also an issue that to open the mobile web browser, the app asks to "Sign in" even though it's signing out, so it's a confusing experience. 

If you want it to behave the same on mobile then [you can disable the expo proxy](https://github.com/jhipster/generator-jhipster-react-native/blob/5759146a2b5e2ff1fb7ab353570a82570fef3484/generators/app/templates/app/config/app-config.js.ejs#L16) and start manually managing your local IPs in your redirect urls.  

I'll see if I can make a direct request to the end_session_endpoint but I don't think that will work.

**Comment by mraible** [comment]
This issue has been solved with v4.2.5 of the React Native blueprint.
